src/daemons/rhsm_d.py

```diff
@@ -187,10 +186,11 @@ def update_status(self, status):
         pre_result = pre_check_status(self.force_signal)
         if pre_result is not None:
             status = pre_result
-        if status != self.last_status:
+        if (status != self.rhsm_icon_cache._read_cache()):
```

Comment:
We should probably go ahead and add a public method for ._read_cache() since we use it externally
in a few places now (repolib, facts)